FreeRTOS/Test/CMock/tasks/tasks_utest_1.c

```diff
@@ -0,0 +1,5117 @@
+    getIddleTaskMemoryValid = true;
+    vTaskStartScheduler();
+    ASSERT_GET_IDLE_TASK_MEMORY_CALLED();
+    TEST_ASSERT_TRUE( xSchedulerRunning );
+    TEST_ASSERT_EQUAL( configINITIAL_TICK_COUNT, xTickCount );
+    TEST_ASSERT_EQUAL( portMAX_DELAY, xNextTaskUnblockTime );
+}
+
+static void block_task( TaskHandle_t task_to_block )
+{
+    TCB_t * tcb_to_block = task_to_block;
+
+    TEST_ASSERT_EQUAL( pxCurrentTCB, task_to_block );
+    uxListRemove_ExpectAndReturn( &tcb_to_block->xStateListItem, 1 );
+    listLIST_ITEM_CONTAINER_ExpectAndReturn( &tcb_to_block->xEventListItem,
+                                             &xSuspendedTaskList );
+    uxListRemove_ExpectAndReturn( &tcb_to_block->xEventListItem, pdTRUE );
+    vListInsertEnd_Expect( &xSuspendedTaskList, &tcb_to_block->xStateListItem );
+    listCURRENT_LIST_LENGTH_ExpectAndReturn( &xSuspendedTaskList,
+                                             uxCurrentNumberOfTasks );
+    vTaskSuspend( task_to_block );
+}
+
+static TaskHandle_t create_task()
+{
+    TaskFunction_t pxTaskCode = NULL;
+    const char * const pcName = { __FUNCTION__ };
+    const uint32_t usStackDepth = 300;
+    void * const pvParameters = NULL;
+    UBaseType_t uxPriority = create_task_priority;
+    TaskHandle_t taskHandle;
+    BaseType_t ret;
+
+    pvPortMalloc_ExpectAndReturn( usStackDepth * sizeof( StackType_t ), stack );
+    pvPortMalloc_ExpectAndReturn( sizeof( TCB_t ), &tcb[ created_tasks ] );
+
+    vListInitialiseItem_Expect( &( tcb[ created_tasks ].xStateListItem ) );
+    vListInitialiseItem_Expect( &( tcb[ created_tasks ].xEventListItem ) );
+    /* TODO: expect set owner */
+    listSET_LIST_ITEM_VALUE_ExpectAnyArgs();
+    /* TODO: expect set owner */
+
+    pxPortInitialiseStack_ExpectAnyArgsAndReturn( stack );
+
+    if( is_first_task )
+    {
+        for( int i = ( UBaseType_t ) 0U; i < ( UBaseType_t ) configMAX_PRIORITIES; i++ )
+        {
+            vListInitialise_ExpectAnyArgs();
+        }
+
+        /* Delayed Task List 1 */
+        vListInitialise_ExpectAnyArgs();
+        /* Delayed Task List 2 */
+        vListInitialise_ExpectAnyArgs();
+        /* Pending Ready List */
+        vListInitialise_ExpectAnyArgs();
+        /* INCLUDE_vTaskDelete */
+        vListInitialise_ExpectAnyArgs();
+        /* INCLUDE_vTaskSuspend */
+        vListInitialise_ExpectAnyArgs();
+        is_first_task = false;
+    }
+
+    vListInsertEnd_ExpectAnyArgs();
+    ret = xTaskCreate( pxTaskCode,
+                       pcName,
+                       usStackDepth,
+                       pvParameters,
+                       uxPriority,
+                       &taskHandle );
+    TEST_ASSERT_EQUAL( pdPASS, ret );
+    ASSERT_SETUP_TCB_CALLED();
+    created_tasks++;
+    return taskHandle;
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/*!
+ * @brief
+ */
+void test_xTaskCreateStatic_null_puxStackBuffer( void )
+{
+    StaticTask_t pxTaskBuffer[ 300 ];
```

Comment:
It's safer to heap allocate so that the address sanitizer can detect out of bounds use.

Reply:
i can't still detect , buffer out of bounds no?

Reply:
When a buffer is stack allocated, that isn't always the case.

Reply:
done..
left the global ones as the sanitizer doc said they catch them